priv/monkey/_next_ls_private_compiler.ex

```diff
@@ -1,5 +1,127 @@
+defmodule NextLSPrivate.DepTracer do
+  @moduledoc false
+
+  @source "dep"
+
+  def trace(:start, _env) do
+    :ok
+  end
+
+  def trace({:alias_reference, meta, module}, env) do
+    parent = parent_pid()
+
+    alias_map = Map.new(env.aliases, fn {alias, mod} -> {mod, alias} end)
+
+    Process.send(
+      parent,
+      {{:tracer, :reference},
+       %{
+         meta: meta,
+         identifier: Map.get(alias_map, module, module),
+         file: env.file,
+         type: :alias,
+         module: module,
+         source: @source
+       }},
+      []
+    )
+
+    :ok
+  end
+
+  def trace({type, meta, module, func, arity}, env) when type in [:remote_function, :remote_macro, :imported_macro] do
+    parent = parent_pid()
+
+    if type == :remote_macro && meta[:closing][:line] != meta[:line] do
+      # this is the case that a macro is getting expanded from inside
+      # another macro expansion
+      :noop
+    else
+      Process.send(
+        parent,
+        {{:tracer, :reference},
+         %{
+           meta: meta,
+           identifier: func,
+           arity: arity,
+           file: env.file,
+           type: :function,
+           module: module,
+           source: @source
+         }},
+        []
+      )
+    end
+
+    :ok
+  end
+
+  def trace({type, meta, func, arity}, env) when type in [:local_function, :local_macro] do
+    parent = parent_pid()
+
+    Process.send(
+      parent,
+      {{:tracer, :reference},
+       %{
+         meta: meta,
+         identifier: func,
+         arity: arity,
+         file: env.file,
+         type: :function,
+         module: env.module,
+         source: @source
+       }},
+      []
+    )
+
+    :ok
+  end
+
+  def trace({:on_module, bytecode, _}, env) do
```

Comment:
i believe all we need is this clause. we don't need to trace references inside dependencies, at least yet.